client/js/components/procedure/StatementSegmentsList/SegmentsRecommendations.vue

```diff
@@ -375,8 +392,22 @@ export default {
   },
 
   mounted () {
-    this.initPagination()
-    this.fetchSegments(this.pagination?.currentPage || 1)
+    // Initialize segment navigation composable
+    const { initializeSegmentPagination } = useSegmentNavigation(
```

Comment:
Is there a specific reason for initializing this twice? I think instead we could add a variable `segmentNavigation` to `data`, then set it here on mounted:
```
this.segmentNavigation = useSegmentNavigation(
      this.statementId,
      this.storageKeyPagination,
      this.pagination,
      this.defaultPagination
)
```
then just use

`this.segmentNavigation.initializeSegmentPagination(...) `
`this.segmentNavigation.calculatePageForSegment(...)`